libgraph/include/katana/TransformationView.h

```diff
@@ -0,0 +1,105 @@
+#ifndef KATANA_LIBGRAPH_KATANA_TRANSFORMATIONVIEW_H_
+#define KATANA_LIBGRAPH_KATANA_TRANSFORMATIONVIEW_H_
+
+#include <memory>
+#include <utility>
+
+#include "katana/PropertyGraph.h"
+
+namespace katana {
+
+/// A TransformationView is a topological transformation of the property graph.
+/// It is derived from the PropertyGraph so that it can be treated by query and
+/// analytics routines as a regular property graph.
+class KATANA_EXPORT TransformationView final : public PropertyGraph {
+public:
+  TransformationView() = default;
+  TransformationView(TransformationView&& other) = default;
+
+  virtual ~TransformationView();
+
+  /// Make a projected graph from a property graph. Shares state with
+  /// the original graph.
+  static std::unique_ptr<TransformationView> MakeProjectedGraph(
+      const PropertyGraph& pg, const std::vector<std::string>& node_types,
+      const std::vector<std::string>& edge_types);
+
+  /// Return the number of nodes of the original property graph.
+  uint64_t NumOriginalNodes() const final {
+    return transformation_.original_to_transformed_nodes_.size();
+  }
+
+  /// Return the number of edges of the original property graph.
+  uint64_t NumOriginalEdges() const final {
+    return transformation_.original_to_transformed_edges_.size();
+  }
```

Comment:
I think it would be better to remove this and instead provide a method to get the original graph and then the user can query it directly. That way there isn't an explosion of "original" methods which exactly mirror the "current graph" methods.

Reply:
IT looks like you mostly have these methods as hooks so that the super class can get the right number for specific operations. If that's the only use, then just making these protected so no one outside the class uses them and expands on them is probably fine.